tests/vao.cpp

```diff
@@ -0,0 +1,100 @@
+/** @file
+ *  Program entry point.
+ *
+ *  @copyright 2006  Palm Studios
+ *
+ *  @author Jesse Buhagiar
+ */
+
+#include "SH3/arc/types.hpp"
+#include "SH3/system/config.hpp"
+#include "SH3/system/exit_code.hpp"
+#include "SH3/system/log.hpp"
+#include "SH3/system/window.hpp"
+#include "SH3/system/glprogram.hpp"
+#include "SH3/graphics/msbmp.hpp"
+#include "SH3/graphics/texture.hpp"
+#include "SH3/system/glbuffer.hpp"
+#include "SH3/system/glvertarray.hpp"
+#include "SH3/types/vertex.hpp"
+#include "SDL2/SDL.h"
+#include <cstdio>
+
+static GLfloat g_vertex_buffer_data[] = {
+   -1.0f, -1.0f, 0.0f,
+   1.0f, -1.0f, 0.0f,
+   0.0f,  1.0f, 0.0f,
+};
+
+static GLfloat glCol[] = {
+    1.0f, 0.0f, 0.0f,
+    0.0f, 1.0f, 0.0f,
+    0.0f, 0.0f, 1.0f,
+};
+
+enum MyAttributes
+{
+    VERTEX_SLOT,
+    COLOR_SLOT,
+    MAX
+};
+
+/**
```

Comment:
I don't think we need to Doxygen-comment our tests.
If we write our own testing framework, then that should be documented, but that's still in the future, if ever.